allium/lib/aroileaders.py

```
@@ -27,9 +27,63 @@
+# Pre-compiled url-token regex for the Option A incomplete-AROI fallback.
+# Matches the CIISS `url:` field token in a raw contact string. We intentionally
+# accept any plausible domain (we are not validating CIISS conformance here —
+# the caller already confirmed AROI parsing FAILED, we just want a recognisable
+# display name).
+_URL_FIELD_FALLBACK_RE = re.compile(
+    r'\burl:(?:https?://)?([^,\s]+)',
+    re.IGNORECASE,
+)
+
+
+def _incomplete_aroi_display_name(contact_info, contact_hash):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_incomplete_aroi_display_name`

(ANN202)